chainservice/builder.go

```diff
@@ -721,7 +727,13 @@ func (builder *Builder) registerRollDPoSProtocol() error {
 	factory := builder.cs.factory
 	dao := builder.cs.blockdao
 	chain := builder.cs.chain
-	getBlockTime := builder.cs.blockTimeCalculator.CalculateBlockTime
+	getBlockTime := func(height uint64) (time.Time, error) {
+		header, err := chain.BlockHeaderByHeight(height)
```

Comment:
why we use `dao` in `registerExecutionProtocol` and `chain` in `registerRollDPoSProtocol`?

Reply:
At present, there is no difference between them, and both will be deprecated in #4590 , instead using the funcs from the blockchain context.